src/essence/Tools/Isochrone/IsochroneTool.js

```diff
@@ -0,0 +1,539 @@
+            [49, 163, 84],
+            [0, 109, 44]
+        ],
+        [ //Blue 4
+            [240, 249, 232],
+            [186, 228, 188],
+            [123, 204, 196],
+            [67, 162, 202],
+            [8, 104, 172]
+        ],
+        [ //Purple 4
+            [237, 248, 251],
+            [179, 205, 227],
+            [140, 150, 198],
+            [136, 86, 167],
+            [129, 15, 124]
+        ]
+    ],
+
+    initialize: function() {
+        this.vars = L_.getToolVars('isochrone');
+
+        //add info for scaling low-res tiles
+        for(const dataType in this.vars.data) {
+            this.dataSources[dataType] = [];
+            for(const src of this.vars.data[dataType]) {
+                const {name, tileurl, minZoom, maxNativeZoom, resolution} = src;
+                let zoomOffset = 0;
+                switch(resolution) {
+                    case 256:
+                        zoomOffset = 0;
+                    break;
+                    case 128:
+                        zoomOffset = 1;
+                    break;
+                    case 64:
+                        zoomOffset = 2;
+                    break;
+                    case 32:
+                        zoomOffset = 3;
+                    break;
+                    default:
+                        console.warn(`IsochroneTool: ${dataType} source "${src.name}" has undefined or nonstandard resolution: ${src.resolution}!`);
+                }
+                
+                const minResolution = Math.max(minZoom - zoomOffset, 0);
+                const maxResolution = maxNativeZoom - zoomOffset;
+
+                //Data source interface! In case this tool gets ts-ified
+                this.dataSources[dataType].push({
+                    name, //string
+                    tileurl, //string
+                    minZoom, //number
+                    maxNativeZoom, //number
+                    resolution, //number
+                    minResolution, //number
+                    maxResolution, //number
+                    zoomOffset //number
+                });
+            }
+        }
+    },
+
+    make: function() {
+        this.MMGISInterface = new interfaceWithMMGIS();
+        
+        $("#iscNew").on("click", () => this.addIsochrone());
+        this.containerEl = $("#isochroneOptionsContainer");
+        this.addIsochrone();
+    },
+
+    destroy: function() {
+        for(const {marker, layerName} of this.managers) {
+            Map_.rmNotNull(marker);
+            Map_.rmNotNull(L_.layersGroup[layerName]);
+        }
+
+        if(this.hoverPolyline !== null) {
+            this.hoverPolyline.remove(Map_.map);
+        }
+        
+        this.managers = [];
+        this.activeManager = null;
+        this.managerCounter = 0;
+        this.MMGISInterface.separateFromMMGIS();
+    },
+
+    getUrlString: function() { //TODO?
+        return '';
+    },
+
+    valueToColor: function(val, rampIndex, steps = 0) {
```

Comment:
for methods/functions that are unique to your tool, I highly recommend including [jsdoc](https://jsdoc.app/)-style function declarations, eg.

```
/**
 * This function does something cool and unique with values and colors.
 */
valueToColor: function(...)
```

VS Code will pick these comments up automatically and display them on-hover. it's a great way to start documenting the intent of your code